packages/esm-service-queues-app/src/patient-queue-header/patient-queue-header.component.tsx

```diff
@@ -15,16 +15,24 @@ import styles from './patient-queue-header.scss';
 
 const PatientQueueHeader: React.FC<{ title?: string }> = ({ title }) => {
   const { t } = useTranslation();
-  const { queueLocations } = useQueueLocations();
+  const { queueLocations, isLoading, error } = useQueueLocations();
   const userSession = useSession();
   const userLocation = userSession?.sessionLocation?.display;
   const currentQueueLocationName = useSelectedQueueLocationName();
 
-  const handleQueueLocationChange = ({ selectedItem }) => {
+  const handleQueueLocationChange = useCallback((selectedItem) => {
     updateSelectedQueueLocationUuid(selectedItem.id);
     updateSelectedQueueLocationName(selectedItem.name);
     updateSelectedServiceName('All');
-  };
+  }, []);
+
+  if (isLoading) {
+    return <p>Loading...</p>; // Show loading state while fetching data
+  }
+
+  if (error) {
+    return <p>Error: {error.message}</p>; // Display error if fetching fails
```

Comment:
Similar issue is in here, not showing the header if there is an error in the request.

Reply:
![image](https://github.com/openmrs/openmrs-esm-patient-management/assets/51502471/0373fb79-689e-4928-b30b-6ef0557e0e64)